### PR TITLE
WIP: Added initial modal window code

### PR DESCRIPTION
**Pull request by alexpricedev**
Modal window with scroll-lock and fade/slide animation

image
image
image
image
image

### TODO

- [x] Close button option with intent colour indicators
- [x] Header styles
- [x] Built-in buttons
- [x] Click background to close

### CODE REVIEW

File: packages/open-ui/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -2,26 +2,26 @@ import React from 'react';
...
+import { IntentOption } from 'interfaces/IntentOption';
 import { SizeOption } from 'interfaces/SizeOption';
 import { getMarginProps } from 'utils/helpers';
+import { toSentenceCase } from 'utils/strings/to-sentence-case';
 import styles from './Button.module.scss';
 
 interface ButtonProps {
-  type?: StyleTypeOption;
+  intent?: IntentOption;
+  onClick: () => void;
```
Comment (gurjit03) [inline packages/open-ui/src/components/Button/Button.tsx:13]: @remotealex lets add a proper type here to catch the event `onClick(e: React.MouseEvent<HTMLButtonElement>) => void`

File: packages/open-ui/src/components/Modal/CloseButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export const CloseButton: React.FC<CloseButtonProps> = ({ close, intent }) => {
+  const cls = classNames(styles.closeButton, {
+    [styles[`intent${toSentenceCase(intent)}`]]: intent,
+  });
+
+  return (
+    <button
+      className={cls}
+      onClick={e => {
```
Comment (gurjit03) [inline packages/open-ui/src/components/Modal/CloseButton.tsx:26]: @remotealex types missing here

File: packages/open-ui/src/components/Modal/Modal.module.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      &:nth-of-type(2) {
+        margin-bottom: 0;
+        margin-left: 2 * $vr;
+      }
+    }
+  }
+}
+
+// Intent colours
+.modalWrapper.intentSuccess {
```
Comment (gurjit03) [inline packages/open-ui/src/components/Modal/Modal.module.scss:93]: @remotealex you should add this under the modalWrapper class above, like currently we are recreating it

File: packages/open-ui/src/components/Modal/CloseButton.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import styles from './CloseButton.module.scss';
+import { IntentOption } from 'interfaces/IntentOption';
+import { toSentenceCase } from 'utils/strings/to-sentence-case';
+
+interface CloseButtonProps {
+  close: () => void;
+  intent: IntentOption;
+}
+
+export const CloseButton: React.FC<CloseButtonProps> = ({ close, intent }) => {
```
Comment (gurjit03) [inline packages/open-ui/src/components/Modal/CloseButton.tsx]: @remotealex why we need to recreate the styles for the closeButton under the Modal CloseButton ?

File: packages/open-ui/src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  text: string;
+}
+
+interface ModalProps {
+  active: boolean;
+  close: () => void;
+  intent?: IntentOption;
+  primaryButton?: ButtonProps;
+  secondaryButton?: ButtonProps;
+  showCloseButton?: boolean;
```
Comment (gurjit03) [inline packages/open-ui/src/components/Modal/Modal.tsx]: @remotealex do we need this prop bro? like if we have secondaryCloseButton we gonna display it else not ?

File: packages/open-ui/src/components/Modal/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import styles from './Modal.module.scss';
+
+interface ButtonProps {
+  onClick: () => void;
+  text: string;
+}
+
+interface ModalProps {
+  active: boolean;
+  close: () => void;
```
Comment (gurjit03) [inline packages/open-ui/src/components/Modal/Modal.tsx]: @remotealex maybe propName as onModalClose --> for explicitely clearing what it is doing.

File: packages/open-ui/src/interfaces/IntentOption.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export type IntentOption =
```
Comment (gurjit03) [inline packages/open-ui/src/interfaces/IntentOption.ts:1]: `IntentOption` --> `IntentOptions` ?

Comment (gurjit03) [reply packages/open-ui/src/interfaces/IntentOption.ts:1]: @remotealex just thinking loud here, is `Primary` or `Secondary` falls into Intent ? Like what I understood is Intent is any action. `Primary` or `Secondary` I am associating with a brand ? Whatcha think ?

**Review by gurjit03 — CHANGES_REQUESTED** [review]

File: packages/open-ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Button/Button.tsx:13]: We're not actually passing the event back to them so this is the correct type

File: packages/open-ui/src/components/Modal/CloseButton.tsx (hunk 2 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/CloseButton.tsx:26]: Not needed as implied from context

File: packages/open-ui/src/components/Modal/Modal.module.scss (hunk 3 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/Modal.module.scss:93]: I prefer to have these custom ones at the bottom. Nest SCSS doesnt actually result in less CSS so I think it's okay like this.

File: packages/open-ui/src/components/Modal/CloseButton.tsx (hunk 4 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/CloseButton.tsx]: From the original `<Button>`? It's different enough to have it's own separate file IMO

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 5 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/Modal.tsx]: I think some people might not want to close button. Maybe we can have it enabled by defauly and change it to `hide`??

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 6 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/Modal.tsx]: `onModalClose` sounds like a callback to me. We could change it to `closeModal` if you want to be verbose?

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 5 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Modal/Modal.tsx]: @remotealex yeah but current execution of code is like if we have secondaryButtonProp passed only then our button is showing

File: packages/open-ui/src/interfaces/IntentOption.ts (hunk 7 above)
Comment (alexpricedev) [reply packages/open-ui/src/interfaces/IntentOption.ts:1]: Singular `IntentOption` makes sense to me in context: 
```
interface ButtonProps {
   intent?: IntentOption;
   ...
```

If we split primary&secondary into its own prop, seperate from `intent`, we'll have to do weird manual overrides. Keeping them in one place makes sense from an implementation perspective. Maybe we can change it from intent to another keyword?

File: packages/open-ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Button/Button.tsx:13]: you mean like we are passing this onClick handler to the users but actual event is not generated here ryt ?

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 5 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Modal/Modal.tsx]: `closeButton` is the (X) in the top right corner...?

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 6 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Modal/Modal.tsx]: @remotealex We are currently passing a callback here to be called when modal is closed ryt ? I am sure I am missing something here bro

File: packages/open-ui/src/components/Button/Button.tsx (hunk 1 above)
Comment (alexpricedev) [reply packages/open-ui/src/components/Button/Button.tsx:13]:
```      
    onClick={e => {
      e.preventDefault();
      e.stopPropagation();
      e.currentTarget.blur();
      onClick();
      return null;
    }}
```
See how we dont pass `e` to their function

Comment (gurjit03) [reply packages/open-ui/src/components/Button/Button.tsx:13]: yeah exactly was mentioning the same

File: packages/open-ui/src/components/Modal/Modal.tsx (hunk 5 above)
Comment (gurjit03) [reply packages/open-ui/src/components/Modal/Modal.tsx]: ahh :( my bad I was mentioning about hte secondaryButton ( as it's was used as close ).

**Pull request merged**